airflow/jobs/job.py

```diff
@@ -252,16 +252,16 @@ def most_recent_job(job_type: str, session: Session = NEW_SESSION) -> Job | None
     :param job_type: job type to query for to get the most recent job for
     :param session: Database session
     """
-    return (
-        session.query(Job)
-        .filter(Job.job_type == job_type)
+    return session.scalars(
```

Comment:
Can this use `session.scalar(...)` instead to eliminate the `first()` call?

Reply:
I wonder why they didn't use scalar in their 1.4/2 examples? scalar worked. Will need to do the same for other places but not sure why they didn't use it as an example

Reply:
I'm not sure we have tests for when the item is more than 1. scalar may order these items randomly and pick the first. WDYT?

Reply:
> scalar may order these items randomly and pick the first

Yeah, but wouldn’t `scalars` and then `first` be subject to the same issue? If I understand things correctly the two are essentially equivalent.

Reply:
I have confirmed why they didn't use it, it's slow on big data. `session.scalar(select(TaskInstance))` got stuck on 17m rows taskinstance

Reply:
Adding a `limit(1)` would fix it, I think?